teammapper-frontend/src/app/core/services/map-sync/map-sync.service.ts

```diff
@@ -429,6 +436,47 @@ export class MapSyncService implements OnDestroy {
       this.mmpService.new(result.map.data, false);
     });
 
+    this.socket.on('mapChangesUndoRedo', (result: ResponseUndoRedoChanges) => {
```

Comment:
same refactoring suggestion I mentioned.

Also: Might be a good place to display a toast? "Change reverted" / "Change re-applied" / "Nodes pasted" something like that? What do you think? I think its helpful for the users to understand what is happening.

Reply:
The question is if it would be too spammy to display toasts, esp. when you have like ~10 users working on the same map. The toasts are quite big and present, so your attention will always be drawn to them - I think quite unhelpful if all they're doing is undo/redo.

Reply:
hm ok we can start without it